src/main/java/de/xm/jdbcexcel/ExcelWriter.java

```diff
@@ -57,129 +57,246 @@ public ExcelWriter(JdbcTemplate template, Map<String, String> replacements) {
+                            columnIndex - 1,
+                            stringCellValue
+                        );
+                    }
+                    return 0;
+
+                case Types.DATE:
+                case Types.TIMESTAMP:
+                case Types.TIMESTAMP_WITH_TIMEZONE:
+                case Types.TIME:
+                case Types.TIME_WITH_TIMEZONE:
+                    Date dateCellValue = rs.getDate(columnIndex);
+
+                    if (!rs.wasNull()) {
+                        return dateCellWriter.writeCellValue(
+                            workbook,
+                            excelRow,
+                            columnIndex - 1,
+                            dateCellValue
+                        );
+                    }
+                    return 0;
+
+                case Types.DOUBLE:
+                case Types.INTEGER:
+                case Types.SMALLINT:
+                case Types.DECIMAL:
+                case Types.FLOAT:
+                case Types.TINYINT:
+                    double doubleCellValue = rs.getDouble(columnIndex);
+
+                    if (!rs.wasNull()) {
+                        return numberCellWriter.writeCellValue(
+                            workbook,
+                            excelRow,
+                            columnIndex - 1,
+                            doubleCellValue
+                        );
+                    }
+                    return 0;
+
+                case Types.BIGINT:
+                case Types.NUMERIC:
+                    BigDecimal bigDecimalCellValue = rs.getBigDecimal(columnIndex);
+
+                    if (!rs.wasNull()) {
+                        return bigDecimalCellWriter.writeCellValue(
+                            workbook,
+                            excelRow,
+                            columnIndex - 1,
+                            bigDecimalCellValue
+                        );
+                    }
+                    return 0;
+
+                case Types.BOOLEAN:
+                case Types.BIT:
+                    boolean booleanCellValue = rs.getBoolean(columnIndex);
+
+                    if (!rs.wasNull()) {
+                        return booleanCellWriter.writeCellValue(
+                            workbook,
+                            excelRow,
+                            columnIndex - 1,
+                            booleanCellValue
+                        );
+                    }
+                    return 0;
+
+                case Types.OTHER:
+                default:
+                    Object objectCellValue = rs.getObject(columnIndex);
+
+                    if (!rs.wasNull()) {
+                        return objectCellWriter.writeCellValue(
+                            workbook,
+                            excelRow,
+                            columnIndex - 1,
+                            objectCellValue
+                        );
+                    }
+                    return 0;
```

Comment:
Zieh doch das ```if (!rs.wasNull()) {``` noch über das Switch

Reply:
Schwierig weil ich direkt mit dem richtigen Typen getAs(blah) aufrufen will (ansonsten gibts ne Exception) und ich eigentlich einen Cast mit getObject vermeiden will

Reply:
ne, das:

```
if (rs.wasNull()) {
   return 0;
}
```

Reply:
Ja das `wasNull` bezieht sich immer auf das letzte `getAs(xyz)`, ich muss erst den getAs aufrufen und kann dann mit `rs.wasNull()` prüfen ob der wert ein Null wert ist. Ist komplett braindead aber leider ists so. Hintergrund ist dass wenn ich z.B. getAsDouble aufrufe auf und in der Sql row `NULL` steht, bekomme ich `0.0` als Ergebnis (???).